gprofiler/profilers/perf.py

```diff
@@ -397,7 +404,7 @@ def __init__(
                 os.path.join(self._profiler_state.storage_dir, "perf.fp"),
                 False,
                 perf_inject,
-                [],
+                pid_args,
```

Comment:
In `PerfProcess`, we always pass `-a` to perf (which means "profile all"). This flag doesn't go with `-p`, I get this warning when I run locally:
```
Warning:
PID/TID switch overriding SYSTEM
```

I suppose we can move the decision of which flag to use into `PerfProcess` - it will get the PID list, then create the `--pid ....` if PIDs are given, or use `-a` if no PIDs are given (default, old behavior)